packages/types/src/sszTypes.ts

```diff
@@ -1,13 +1,192 @@
+export const allForksBlobs = pick(typesByFork, ...blobsForks);
+export const allForksBlinded = {
+  bellatrix: {
+    BeaconBlockBody: bellatrix.BlindedBeaconBlockBody,
+    BeaconBlock: bellatrix.BlindedBeaconBlock,
+    SignedBeaconBlock: bellatrix.SignedBlindedBeaconBlock,
+  },
+  capella: {
+    BeaconBlockBody: capella.BlindedBeaconBlockBody,
+    BeaconBlock: capella.BlindedBeaconBlock,
+    SignedBeaconBlock: capella.SignedBlindedBeaconBlock,
+  },
+  deneb: {
+    BeaconBlockBody: deneb.BlindedBeaconBlockBody,
+    BeaconBlock: deneb.BlindedBeaconBlock,
+    SignedBeaconBlock: deneb.SignedBlindedBeaconBlock,
+  },
+};
+
+// TODO: These helpers should be removed along with `allForksBlinded`
+type SSZBlindedTypesByFork = {
+  [F in keyof typeof allForksBlinded]: {
+    [T in keyof (typeof allForksBlinded)[F]]: (typeof allForksBlinded)[F][T];
+  };
+};
+
+// TODO: These helpers should be removed along with `allForksBlinded`
+export type SSZBlindedTypesFor<
+  F extends ForkExecution,
+  K extends keyof SSZBlindedTypesByFork[F] | void = void,
+> = K extends void
+  ? // It compiles fine, need to debug the error
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-expect-error
+    {[K2 in keyof SSZBlindedTypesByFork[F]]: UnionForksTypeOf<SSZBlindedTypesByFork[F][K2]>}
+  : // It compiles fine, need to debug the error
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-expect-error
+    UnionForksTypeOf<SSZBlindedTypesByFork[F][Exclude<K, void>]>;
+
+/**
+ * A type of union of forks must accept as any parameter the UNION of all fork types.
+ */
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+type UnionForksTypeOf<UnionOfForkTypes extends ContainerType<any>> = CompositeType<
+  ValueOf<UnionOfForkTypes>,
+  CompositeView<UnionOfForkTypes>,
+  CompositeViewDU<UnionOfForkTypes>
+>;
+
+type SSZTypesByFork = {
+  [F in keyof typeof typesByFork]: {
+    [T in keyof (typeof typesByFork)[F]]: (typeof typesByFork)[F][T];
+  };
+};
+
+export type SSZTypesFor<F extends ForkName, K extends keyof SSZTypesByFork[F] | void = void> = K extends void
+  ? // It compiles fine, need to debug the error
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-expect-error
+    {[K2 in keyof SSZTypesByFork[F]]: UnionForksTypeOf<SSZTypesByFork[F][K2]>}
+  : // It compiles fine, need to debug the error
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-expect-error
+    UnionForksTypeOf<SSZTypesByFork[F][Exclude<K, void>]>;
+
+export function sszTypesFor<F extends ForkName, K extends keyof SSZTypesByFork[F] | void = void>(
+  fork: F,
+  typeName?: K
+): SSZTypesFor<F, K> {
+  return (
+    typeName === undefined ? typesByFork[fork] : typesByFork[fork][typeName as keyof SSZTypesByFork[F]]
+  ) as SSZTypesFor<F, K>;
+}
+
+export type SSZInstanceTypesFor<F extends ForkName, K extends keyof SSZTypesByFork[F]> = CompositeViewDU<
```

Comment:
I don't think this type is necessary or helpful.
Better for consumer to do `CompositeViewDU<SSZTypesFor<F, T>>`

eg: `type BeaconStateAllForks = CompositeViewDU<SSZTypesFor<ForkAll, "BeaconState">>`